example/generator/odp_generator.c

```diff
@@ -1210,10 +1214,35 @@ int main(int argc, char *argv[])
 	/* Configure scheduler */
 	odp_schedule_config(NULL);
 
+	if (odp_pool_capability(&pool_capa)) {
+		ODPH_ERR("Error: pool capability failed\n");
+		return -1;
+	}
+
+	pkt_len = (args->appl.payload + MAX_HDR_LEN) > POOL_PKT_LEN ?
+			(args->appl.payload + MAX_HDR_LEN) : POOL_PKT_LEN;
+
+	if (pool_capa.pkt.max_len && pkt_len > pool_capa.pkt.max_len)
+		pkt_len = pool_capa.pkt.max_len;
+
+	/* Check weather we have sufficient segments to support requested packet
+	 * length, if not adjust to bigger segment size */
+	if ((seg_len * pool_capa.pkt.max_segs_per_pkt) < pkt_len)
+		seg_len = pkt_len / pool_capa.pkt.max_segs_per_pkt;
+
+	if (pool_capa.pkt.min_seg_len && seg_len < pool_capa.pkt.min_seg_len)
+		seg_len = pool_capa.pkt.min_seg_len;
```

Comment:
This is not needed. min_seg_len is not the minimum allowed value for the seg_len parameter but just indicates that the segments will not be shorter than that even if an application does not request that long segments.

Reply:
Will remove this code